src/Fixer/Import/FullyQualifiedStrictTypesFixer.php

```diff
@@ -245,14 +298,16 @@ private function fixPhpDoc(Tokens $tokens, int $index, array $uses, string $name
      */
     private function fixExtendsImplements(Tokens $tokens, int $index, array $uses, string $namespaceName): void
     {
+        // We handle `extends` and `implements` with similar logic, but we need to exit the loop under different conditions.
+        $isExtends = $tokens[$index]->equals([T_EXTENDS]);
         $index = $tokens->getNextMeaningfulToken($index);
         $extend = ['content' => '', 'tokens' => []];
 
         while (true) {
             if ($tokens[$index]->equalsAny([',', '{', [T_IMPLEMENTS]])) {
                 $this->shortenClassIfPossible($tokens, $extend, $uses, $namespaceName);
 
-                if ($tokens[$index]->equals('{')) {
+                if ($tokens[$index]->equalsAny($isExtends ? [[T_IMPLEMENTS], '{'] : ['{'])) {
```

Comment:
FYI: basically it is a fix for redundant analysis that is currently executed  when `extends` and `implements` are both used in a class. This method is invoked 2 times in `applyFix()` because both `T_EXTENDS` and  `T_IMPLEMENTS` trigger it, but currently both are exiting on `{`, which means `implements` part is processed 2 times.

In the context of this particular PR, it resulted in symbols being imported twice, because e.g. `implements \Other\Foo` was shortened to `implements Foo` in the first run (for `extends`), and when processing `implements` it was registering `Foo` for importing. I've fixed the issue of importing non-FQNs (symbol must start with `\` to be registered for importing), but this scenario allowed me to find the redundant loop.